src/rdkafka_offset.c

```diff
@@ -812,6 +818,10 @@ void rd_kafka_offset_reset (rd_kafka_toppar_t *rktp, int64_t err_offset,
                      rd_kafka_offset2str(offset),
                      reason, rd_kafka_err2str(err));
 
+        /* Note: If rktp is not delegated to the leader, then low and high
+           offsets will necessarily be cached from the last FETCH request,
+           and so this offset query will never occur in that case for
+           BEGINNING / END logical offsets. */
```

Comment:
This is only true of the partition has been fetched at least once from the follower, right? So is there a  race condition here between the delegate-to-follower and first-valid-fetch?